BioMed Central.js

```diff
@@ -0,0 +1,258 @@
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+    ***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+    if (url.indexOf('biomedcentral.com/articles/10.1186/')>-1) {
+        return "journalArticle";
+    } else if (getSearchResults(doc, true)) {
+        return "multiple";
+    }
+}
+
+function getSearchResults(doc, checkOnly) {
+    var items = {};
+    var found = false;
+    var rows = ZU.xpath(doc, '//div[@id="search-container"]//article//h3[contains(@class,"ResultsList_title")]/a');
+    for (var i=0; i<rows.length; i++) {
+        var href = rows[i].href;
+        var title = ZU.trimInternal(rows[i].textContent);
+        if (!href || !title) continue;
+        if (checkOnly) return true;
+        found = true;
+        items[href] = title;
+    }
+    return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+    if (detectWeb(doc, url) == "multiple") {
+        Zotero.selectItems(getSearchResults(doc, false), function (items) {
+            if (!items) {
+                return true;
+            }
+            var articles = new Array();
+            for (var i in items) {
+                articles.push(i);
+            }
+            ZU.processDocuments(articles, scrape);
+        });
+    } else {
+        scrape(doc, url);
+    }
+}
+
+function scrape(doc, url) {
+	var DOI = url.match(/\/(10\.[^#?]+)/)[1];
+	var risURL = "http://citation-needed.services.springer.com/v2/references/" + DOI + "?format=refman&flavour=citation";
+	//Z.debug(risURL)
+	
+	var pdfURL = doc.getElementById("articlePdf");
+	//Z.debug("pdfURL: " + pdfURL);
+	ZU.doGet(risURL, function(text) {
+		var translator = Zotero.loadTranslator("import");
+		translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
+		translator.setString(text);
+		translator.setHandler("itemDone", function(obj, item) {
+
+			//We have to fix issue, pages from some wrong RIS data.
+			var citation = ZU.xpath(doc, '//span[@class="ArticleCitation_Volume"]');
+			if (citation.length>0) {
+				fixCitation(item, citation[0].innerHTML);
+			}
+
+			var keywordsNodes = doc.getElementsByClassName("Keyword");
+			for (var i=0; i<keywordsNodes.length; i++) {
+				item.tags.push( keywordsNodes[i].textContent );
+			}
+			
+			if (pdfURL) {
+				item.attachments.push({
+					url: pdfURL.href,
+					title: "Full Text PDF",
+					mimeType: "application/pdf"
+				});
+			}
+			item.attachments.push({
+				title: "Snapshot",
+				document: doc
+			});
+			item.complete();
+		})
+		translator.translate();
+	})
+}
+
+function fixCitation(item, citation) {
+	//e.g. citation = <strong>8</strong>:212
+	//  or citation = <strong>2</strong>(1):S1
+	var re = /<strong>(\d+)<\/strong>([\w()]*):(\w+)/;
+	var m = citation.match(re);
+	//Z.debug(m);
+	if (m) {
+		//save the pages (PDF) somewhere first
```

Comment:
could we
a) include a test of this and
b) test for the presence of volume and issue before overwriting it?